### ISSUE TITLE
"set pen trails to" block limits the space that the pen can draw on to the extent of the "trails"

### ISSUE THREAD

**Issue by eprebys** [jmoenig/Snap--Build-Your-Own-Blocks#1352]
When the "set pen trails to" block is used, it limits the space that the pen can draw to the extent of the input to "set pen trails to". This limitation continues until the pen marks are cleared from the canvas.

As an example, if I make a 240x180 design on the screen using the pen, then "set pen trails to" that "trails", I will thereafter only be able to draw in the top left quadrant of the stage.

![image](https://cloud.githubusercontent.com/assets/2782215/17279260/0564da5c-573e-11e6-9b14-7eba9078bc93.png)

**Comment by jmoenig** [comment]
Hmm... isn't that what you'd expect when you set the pen trails to something smaller than the stage? If you just want to display some picture you can stamp it, you know, and that won't limit anything.

**Comment by eprebys** [comment]
A primary use case I envision for pen trails is creating an image editing app with save and restore functionality. Maybe this is more of an issue with "pen trails" than with "set pen trails to". If "pen trails" was a capture of the full canvas (or if it took arguments) this wouldn't be an issue.

**Comment by cycomachead** [comment]
I think of this as a use case to. I also think that setting to a previous state wouldn't change the "drawable" space, since the previous pen trails could use the entire stage. (or rather, previous if you went to the first case.)

**Comment by jmoenig** [comment]
Yes. Now I remember this being the use case I envisioned originally :-) I later changed it so that the ```pen trails``` reporter auto-crops, so you can let a sprite wear the pen-trails and so the rotation center gets right. I can think of several ways to address this issue:

* add a new block for cropping
* always auto-crop when wearing a costume
* add a Boolean cropping switch to the ```pen trails``` reporter

Any other? I don't really like any of these.

Oh wait, there's one more I'm seriously thinking about:

* *remove* the ```set pen trails``` command

Instead, if you want to remember the state of the pentrails you save them to a costume using the ```pen trails``` reporter, and when you want to restore them you ```clear``` the pen trails and ```stamp``` the saved pen trails with a helper sprite. Would that work?

**Comment by brianharvey** [comment]
I definitely vote no on auto-cropping costumes, at least unless you take care to include the rotation center in the bounding rectangle.  I fairly often set a rotation point outside the visible costume.

I don't see why PEN TRAILS can't auto-crop without also restricting the bounds of later drawing.  We don't even restrict the bounds to the size of the stage!  We provide a window into an infinite drawing plane.
